pkg/ccl/storageccl/revision_reader.go

```diff
@@ -31,48 +31,57 @@ type VersionedValues struct {
 func GetAllRevisions(
 	ctx context.Context, db *kv.DB, startKey, endKey roachpb.Key, startTime, endTime hlc.Timestamp,
 ) ([]VersionedValues, error) {
-	// TODO(dt): version check.
-	header := roachpb.Header{Timestamp: endTime}
-	req := &roachpb.ExportRequest{
-		RequestHeader: roachpb.RequestHeader{Key: startKey, EndKey: endKey},
-		StartTime:     startTime,
-		MVCCFilter:    roachpb.MVCCFilter_All,
-		ReturnSST:     true,
-	}
-	resp, pErr := kv.SendWrappedWith(ctx, db.NonTransactionalSender(), header, req)
-	if pErr != nil {
-		return nil, pErr.GoError()
-	}
-
 	var res []VersionedValues
-	for _, file := range resp.(*roachpb.ExportResponse).Files {
-		iter, err := storage.NewMemSSTIterator(file.SST, false)
-		if err != nil {
-			return nil, err
+	for {
+		// TODO(dt): version check.
+		header := roachpb.Header{Timestamp: endTime, TargetBytes: 1}
+		req := &roachpb.ExportRequest{
+			RequestHeader:  roachpb.RequestHeader{Key: startKey, EndKey: endKey},
+			StartTime:      startTime,
+			MVCCFilter:     roachpb.MVCCFilter_All,
+			ReturnSST:      true,
+			TargetFileSize: 1 << 20,
```

Comment:
This number IMHO asks for a comment or constant why we do 1M file sizes. Even if it is just arbirtary number that is small/big enough.